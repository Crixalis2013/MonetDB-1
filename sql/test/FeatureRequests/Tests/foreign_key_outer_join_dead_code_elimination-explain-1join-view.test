statement ok
set optimizer = 'sequential_pipe'

query T rowsort
explain select count(*) from v1
----
33 values hashing to 2b8c5417de37fe80bfb4c29efb2821a4

query T rowsort
explain select id       from v1 order by id
----
42 values hashing to e6e593febde8f49a3c0508b264391c8b

query T rowsort
explain select id , v1  from v1 order by id
----
63 values hashing to 095c8cbb12305fb10b4bce207a971e20

statement ok
set optimizer = 'default_pipe'


