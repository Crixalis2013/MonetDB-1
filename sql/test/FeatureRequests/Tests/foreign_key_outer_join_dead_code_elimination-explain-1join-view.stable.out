stdout of test 'foreign_key_outer_join_dead_code_elimination-explain-1join-view` in directory 'sql/test/FeatureRequests` itself:


# 01:44:33 >  
# 01:44:33 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=31552" "--set" "mapi_usock=/var/tmp/mtest-6324/.s.monetdb.31552" "--set" "monet_prompt=" "--forcemito" "--dbpath=/ufs/manegold/_/Monet/default/prefix/_/var/MonetDB/mTests_sql_test_FeatureRequests" "--set" "embedded_r=yes" "--set" "embedded_py=true"
# 01:44:33 >  

# MonetDB 5 server v11.28.0
# This is an unreleased version
# Serving database 'mTests_sql_test_FeatureRequests', using 4 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 31.309 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2018 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://earth.da.cwi.nl:31552/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-6324/.s.monetdb.31552
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/Python2 module loaded
# MonetDB/R   module loaded


# 01:44:33 >  
# 01:44:33 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-6324" "--port=31552"
# 01:44:33 >  

#set optimizer = 'sequential_pipe';

#explain select count(*) from v1;
% .explain # table_name
% mal # name
% clob # type
% 103 # length
function user.main():void;
    X_1:void := querylog.define("explain select count(*) from v1;":str, "sequential_pipe":str, 13:int);
    X_4:int := sql.mvc();
    C_5:bat[:oid] := sql.tid(X_4:int, "sys":str, "fk":str);
    X_20:lng := aggr.count(C_5:bat[:oid]);
    sql.resultSet(".%1":str, "%1":str, "bigint":str, 64:int, 0:int, 7:int, X_20:lng);
end user.main;
#inline               actions= 0 time=0 usec 
#remap                actions= 0 time=1 usec 
#costmodel            actions= 1 time=1 usec 
#coercion             actions= 0 time=1 usec 
#evaluate             actions= 0 time=2 usec 
#emptybind            actions= 1 time=4 usec 
#pushselect           actions= 0 time=2 usec 
#aliases              actions= 1 time=3 usec 
#mergetable           actions= 0 time=16 usec 
#deadcode             actions= 2 time=4 usec 
#aliases              actions= 0 time=0 usec 
#constants            actions= 0 time=2 usec 
#commonTerms          actions= 0 time=2 usec 
#projectionpath       actions= 0 time=1 usec 
#reorder              actions= 1 time=11 usec 
#deadcode             actions= 0 time=3 usec 
#matpack              actions= 0 time=0 usec 
#multiplex            actions= 0 time=1 usec 
#profiler             actions=1 time=0 usec 
#candidates           actions=1 time=0 usec 
#postfix              actions= 1 time=2 usec 
#deadcode             actions= 1 time=3 usec 
#wlc                  actions= 0 time=0 usec 
#garbagecollector     actions= 1 time=19 usec 
#total                actions=26 time=130 usec 

#explain select id       from v1 order by id;
% .explain # table_name
% mal # name
% clob # type
% 126 # length
function user.main():void;
    X_1:void := querylog.define("explain select id from v1 order by id;":str, "sequential_pipe":str, 19:int);
    X_27:bat[:str] := bat.pack("sys.v1":str);
    X_28:bat[:str] := bat.pack("id":str);
    X_29:bat[:str] := bat.pack("int":str);
    X_30:bat[:int] := bat.pack(32:int);
    X_31:bat[:int] := bat.pack(0:int);
    X_4:int := sql.mvc();
    C_5:bat[:oid] := sql.tid(X_4:int, "sys":str, "fk":str);
    X_15:bat[:int] := sql.bind(X_4:int, "sys":str, "fk":str, "id":str, 0:int);
    X_19:bat[:int] := algebra.projection(C_5:bat[:oid], X_15:bat[:int]);
    (X_20:bat[:int], X_21:bat[:oid]) := algebra.sort(X_19:bat[:int], false:bit, false:bit, false:bit);
    X_25:bat[:int] := algebra.projection(X_21:bat[:oid], X_19:bat[:int]);
    sql.resultSet(X_27:bat[:str], X_28:bat[:str], X_29:bat[:str], X_30:bat[:int], X_31:bat[:int], X_25:bat[:int]);
end user.main;
#inline               actions= 0 time=1 usec 
#remap                actions= 0 time=1 usec 
#costmodel            actions= 1 time=0 usec 
#coercion             actions= 0 time=1 usec 
#evaluate             actions= 0 time=3 usec 
#emptybind            actions= 1 time=5 usec 
#pushselect           actions= 0 time=3 usec 
#aliases              actions= 1 time=4 usec 
#mergetable           actions= 0 time=20 usec 
#deadcode             actions= 2 time=6 usec 
#aliases              actions= 0 time=0 usec 
#constants            actions= 1 time=4 usec 
#commonTerms          actions= 0 time=3 usec 
#projectionpath       actions= 0 time=2 usec 
#reorder              actions= 1 time=16 usec 
#deadcode             actions= 0 time=4 usec 
#matpack              actions= 0 time=0 usec 
#multiplex            actions= 0 time=1 usec 
#profiler             actions=1 time=1 usec 
#candidates           actions=1 time=1 usec 
#postfix              actions= 0 time=1 usec 
#deadcode             actions= 0 time=4 usec 
#wlc                  actions= 0 time=0 usec 
#garbagecollector     actions= 1 time=24 usec 
#total                actions=26 time=156 usec 

#explain select id , v1  from v1 order by id;
% .explain # table_name
% mal # name
% clob # type
% 130 # length
function user.main():void;
    X_1:void := querylog.define("explain select id , v1 from v1 order by id;":str, "sequential_pipe":str, 58:int);
    X_48:bat[:int] := bat.new(nil:int);
    X_4:int := sql.mvc();
    C_5:bat[:oid] := sql.tid(X_4:int, "sys":str, "fk":str);
    X_19:bat[:oid] := sql.bind_idxbat(X_4:int, "sys":str, "fk":str, "fk_fk1_fkey":str, 0:int);
    (X_21:bat[:oid], X_22:bat[:oid]) := sql.bind_idxbat(X_4:int, "sys":str, "fk":str, "fk_fk1_fkey":str, 2:int);
    X_23:bat[:oid] := sql.delta(X_19:bat[:oid], X_21:bat[:oid], X_22:bat[:oid]);
    X_37:bat[:oid] := algebra.projection(C_5:bat[:oid], X_23:bat[:oid]);
    C_24:bat[:oid] := sql.tid(X_4:int, "sys":str, "pk1":str);
    (X_39:bat[:oid], X_40:bat[:oid]) := algebra.join(X_37:bat[:oid], C_24:bat[:oid], nil:BAT, nil:BAT, false:bit, nil:lng);
    X_15:bat[:int] := sql.bind(X_4:int, "sys":str, "fk":str, "id":str, 0:int);
    X_36:bat[:int] := algebra.projection(C_5:bat[:oid], X_15:bat[:int]);
    X_47:bat[:int] := algebra.projection(X_39:bat[:oid], X_36:bat[:int]);
    C_45:bat[:oid] := bat.mirror(X_36:bat[:int]);
    C_46:bat[:oid] := algebra.difference(C_45:bat[:oid], X_39:bat[:oid], nil:BAT, nil:BAT, false:bit, false:bit, nil:lng);
    X_31:bat[:int] := sql.bind(X_4:int, "sys":str, "pk1":str, "v1":str, 0:int);
    X_50:bat[:int] := bat.append(X_48:bat[:int], X_47:bat[:int], true:bit);
    X_52:bat[:int] := algebra.projection(C_46:bat[:oid], X_36:bat[:int]);
    X_53:bat[:int] := bat.append(X_50:bat[:int], X_52:bat[:int], true:bit);
    X_61:bat[:int] := bat.new(nil:int);
    X_60:bat[:int] := algebra.projectionpath(X_40:bat[:oid], C_24:bat[:oid], X_31:bat[:int]);
    X_62:bat[:int] := bat.append(X_61:bat[:int], X_60:bat[:int], true:bit);
    X_64:bat[:int] := algebra.project(C_46:bat[:oid], nil:int);
    X_65:bat[:int] := bat.append(X_62:bat[:int], X_64:bat[:int], true:bit);
    X_78:bat[:str] := bat.pack(".v1":str, ".v1":str);
    X_79:bat[:str] := bat.pack("id":str, "v1":str);
    X_80:bat[:str] := bat.pack("int":str, "int":str);
    X_81:bat[:int] := bat.pack(32:int, 32:int);
    X_82:bat[:int] := bat.pack(0:int, 0:int);
    (X_72:bat[:int], X_73:bat[:oid]) := algebra.sort(X_53:bat[:int], false:bit, false:bit, false:bit);
    X_75:bat[:int] := algebra.projection(X_73:bat[:oid], X_53:bat[:int]);
    X_76:bat[:int] := algebra.projection(X_73:bat[:oid], X_65:bat[:int]);
    sql.resultSet(X_78:bat[:str], X_79:bat[:str], X_80:bat[:str], X_81:bat[:int], X_82:bat[:int], X_75:bat[:int], X_76:bat[:int]);
end user.main;
#inline               actions= 0 time=1 usec 
#remap                actions= 0 time=2 usec 
#costmodel            actions= 1 time=1 usec 
#coercion             actions= 2 time=7 usec 
#evaluate             actions= 0 time=5 usec 
#emptybind            actions= 2 time=8 usec 
#pushselect           actions= 0 time=2 usec 
#aliases              actions= 2 time=7 usec 
#mergetable           actions= 0 time=36 usec 
#deadcode             actions=14 time=10 usec 
#aliases              actions= 0 time=1 usec 
#constants            actions= 5 time=6 usec 
#commonTerms          actions= 0 time=5 usec 
#projectionpath       actions= 1 time=8 usec 
#reorder              actions= 1 time=27 usec 
#deadcode             actions= 1 time=7 usec 
#matpack              actions= 0 time=0 usec 
#multiplex            actions= 0 time=1 usec 
#profiler             actions=1 time=2 usec 
#candidates           actions=1 time=1 usec 
#postfix              actions= 0 time=0 usec 
#deadcode             actions= 0 time=6 usec 
#wlc                  actions= 0 time=0 usec 
#garbagecollector     actions= 1 time=24 usec 
#total                actions=26 time=156 usec 

# 01:44:33 >  
# 01:44:33 >  "Done."
# 01:44:33 >  

